mergify_engine/worker.py

```diff
@@ -235,18 +237,53 @@ async def run_engine(
         logger.debug("engine in thread end")
 
 
+class BucketSelector(typing.Protocol):
+    async def next_org_bucket(self) -> typing.Optional[OrgBucketNameType]:
+        ...
+
+
+@dataclasses.dataclass
+class DedicatedOrgBucketSelector:
+    redis_stream: utils.RedisStream
+    owner_id: github_types.GitHubAccountIdType
+
+    def _is_org_bucket_for_me(self, org_bucket: bytes) -> bool:
+        owner_id = int(org_bucket.split(b"~")[1])
+        return owner_id == self.owner_id
+
+    async def next_org_bucket(self) -> typing.Optional[OrgBucketNameType]:
+        # NOTE(sileht): Should we drop the 30s latency ? That may impact the rate limit...
+        now = time.time()
+        for org_bucket in await self.redis_stream.zrangebyscore(
+            "streams",
+            min=0,
+            max=now,
+        ):
```

Comment:
That looks suboptimal.
What about using directly `ZSCORE streams org_bucket` and checking the returned value?

Reply:
Ok so maybe just add a note at least here 😢